ui-v2/src/components/deployments/deployment-details-upcoming-tab.tsx

```diff
@@ -39,62 +40,96 @@ export const DeploymentDetailsUpcomingTab = ({
 			},
 			operator: "and_",
 		},
-		limit: pagination.pageSize,
-		page: pagination.pageIndex + 1, // + 1 for to account for react table's 0 index
+		limit: pagination.limit,
+		page: pagination.page,
 		sort,
 	});
 
+	const handleResetFilters = () => {
+		resetFilters();
+		setSelectedRows(new Set());
+	};
+
+	const addRow = (id: string) =>
+		setSelectedRows((curr) => new Set([...Array.from(curr), id]));
+
+	const removeRow = (id: string) =>
+		setSelectedRows(
+			(curr) => new Set([...Array.from(curr).filter((i) => i !== id)]),
+		);
```

Comment:
I'd probably avoid going between arrays and sets like this. Add is simpler than remove because `Set.add()` returns the set but `Set.delete()` returns a boolean. 

```suggestion
	const addRow = (id: string) =>
		setSelectedRows((curr) => new Set(curr).add(id));

	const removeRow = (id: string) =>
		setSelectedRows((curr) => {
		const newValue = new Set(curr)
		newValue.delete(id)
		return newValue
		});
```

Reply:
Thanks for the snippets!